ground-station-openmct/servers/balloon.js

```diff
@@ -60,13 +66,18 @@ Balloon.prototype.pull = function () {
     fetch(`http://127.0.0.1:8000/data/tel/${telCounter}`)
     .then(response => response.json())
     .then(data => {
-      console.log(data);
-      temp = data.temperature;
-      pressure = data.pressure;
-      humidity = data.humidity;
-      x = data.x_rotation;
-      y = data.y_rotation;
-      z = data.z_rotation;
+        console.log(data);
+        temp = data.temperature;
+        pressure = data.pressure;
+        humidity = data.humidity;
+        x = data.x_rotation;
+        y = data.y_rotation;
+        z = data.z_rotation;
+        no_data = false;
+        console.log(data.temperature  == undefined);
+        if (data.temperature == undefined) {
```

Comment:
why does every single data point need to be defined in order to update state / telemetry?  What if instead you skipped undefined data points in the forEach loop in generateTelemetry?  That way we handle undefined values for all data points, not just temp and latitude, while at the same time allowing updating the rest of the data that is not undefined